packages/block-library/src/post-content/index.php

```diff
@@ -0,0 +1,36 @@
+<?php
+/**
+ * Server-side rendering of the `core/post-content` block.
+ *
+ * @package WordPress
+ */
+
+/**
+ * Renders the `core/post-content` block on the server.
+ *
+ * @return string Returns the filtered post content of the current post.
+ */
+function render_block_core_post_content() {
+	// TODO: Without this temporary fix, an infinite loop can occur.
+	if ( is_admin() || defined( 'REST_REQUEST' ) ) {
```

Comment:
The logic of the condition is not clear. maybe expand on the comment. Also if this is temporary what's the ideal fix?

Reply:
@felixarntz It's because in the admin views and in REST requests there is no post loop right? So the rest of this function becomes infinitely recursive?

Is there even a fix for this?

Reply:
I suppose this should be necessary for all post related blocks too. 

For me, this is another reason the Post block is clearer as the condition becomes "is there a parent post context".

Reply:
That wouldn't solve this and there is not currently a way for a parent post block to provide context for children during server rendering. You would be relying on the order in which the block tree is traversed while serializing which would be extremely fragile, specially when you start nesting post contexts.

Solving that shouldn't block this PR.